src/rgw/rgw_lc_s3.cc

```diff
@@ -137,6 +139,35 @@ void LCFilter_S3::decode_xml(XMLObj *obj)
    * Empty filters are allowed:
    * https://docs.aws.amazon.com/AmazonS3/latest/dev/intro-lifecycle-rules.html
    */
+  if (XMLObj* and_xml = obj->find_first("And"); and_xml) {
```

Comment:
if we're going to do all this, should we consider an actual parse of the xml document?